packages/core/src/App/Containers/notification-messages.jsx

```diff
@@ -1,36 +1,66 @@
-                .filter(message => !marked_notifications.includes(message.key))
-                .slice(0, max_display_notifications)
-                .map((notification, idx) => (
-                    <CSSTransition
-                        appear
-                        key={idx}
-                        in={!!notification.header}
-                        timeout={150}
-                        classNames={{
-                            appear: 'notification--enter',
-                            enter: 'notification--enter',
-                            enterDone: 'notification--enter-done',
-                            exit: 'notification--exit',
-                        }}
-                        unmountOnExit
-                    >
-                        <Notification data={notification} removeNotificationMessage={removeNotificationMessage} />
-                    </CSSTransition>
-                ))}
-        </TransitionGroup>
-    </div>
-);
+class NotificationMessages extends React.Component {
+    state = {};
+    setRef = el => {
+        if (el && el.parentElement) {
+            this.setState({
+                bounds: el.parentElement.getBoundingClientRect(),
+            });
+        }
+    };
+
+    render() {
+        const { marked_notifications, notification_messages, removeNotificationMessage } = this.props;
+        const { bounds } = this.state;
+        const style = {
+            top: bounds && bounds.top + 8,
+        };
+
+        const portal = ReactDOM.createPortal(
+            <div className='notification-messages' style={style}>
+                <TransitionGroup component='div'>
+                    {notification_messages
+                        .filter(message => !marked_notifications.includes(message.key))
+                        .slice(0, max_display_notifications)
+                        .map((notification, idx) => (
+                            <CSSTransition
+                                appear
+                                key={idx}
+                                in={!!notification.header}
+                                timeout={150}
+                                classNames={{
+                                    appear: 'notification--enter',
+                                    enter: 'notification--enter',
+                                    enterDone: 'notification--enter-done',
+                                    exit: 'notification--exit',
+                                }}
+                                unmountOnExit
+                            >
+                                <Notification
+                                    data={notification}
+                                    removeNotificationMessage={removeNotificationMessage}
+                                />
+                            </CSSTransition>
+                        ))}
+                </TransitionGroup>
+            </div>,
+            document.getElementById('popup_root')
```

Comment:
As discussed, `popup_root` will be removed since we can use `modal_root` if we want the Toast Message to be rendered above of `Modal` or `Dialog` based components.

Reply:
Done 👍